src/sentry/dynamic_sampling/tasks/common.py

```diff
@@ -345,6 +404,169 @@ def get_active_orgs_with_projects_counts(
+                        Column("org_id"),
+                    ],
+                    where=[
+                        Condition(
+                            # Column("timestamp"), Op.GTE, datetime.utcnow() - self.time_interval
+                            Column("timestamp"),
+                            Op.GTE,
+                            datetime.utcnow() - timedelta(hours=1),
+                        ),
+                        Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                        Condition(Column("metric_id"), Op.EQ, self.metric_id),
+                    ],
+                    granularity=self.granularity,
+                    orderby=[
+                        OrderBy(Column("org_id"), Direction.ASC),
+                    ],
+                )
+                .set_limit(CHUNK_SIZE + 1)
+                .set_offset(self.offset)
+            )
+            request = Request(
+                dataset=Dataset.PerformanceMetrics.value, app_id="dynamic_sampling", query=query
+            )
+            self.log_state.num_db_calls += 1
+            data = raw_snql_query(
+                request,
+                referrer=Referrer.DYNAMIC_SAMPLING_COUNTERS_GET_ORG_TRANSACTION_VOLUMES.value,
+            )["data"]
+            count = len(data)
+
+            self.has_more_results = count > CHUNK_SIZE
+            self.offset += CHUNK_SIZE
+            if self.has_more_results:
+                data = data[:-1]
+            self.log_state.num_rows_total += len(data)
+            for row in data:
+                self.last_result.append(
+                    OrganizationDataVolume(
+                        org_id=row["org_id"], total=row["total_count"], indexed=row["keep_count"]
+                    )
+                )
+
+        if len(self.last_result) > 0:
+            # we have some data left return up to the max amount
+            return self._get_from_cache()  # we still have something left in cache
+        else:
+            # nothing left in the DB or cache
+            raise StopIteration()
+
+    def get_current_state(self):
+        """
+        Returns the current state of the iterator (how many orgs and projects it has iterated over)
+
+        part of the ContexIterator protocol
+
+        """
+        return self.log_state
+
+    def set_current_state(self, log_state: DynamicSamplingLogState):
+        self.log_state = log_state
+
+    def _enough_results_cached(self):
+        """
+        Return true if we have enough data to return a full batch in the cache (i.e. last_result)
+        """
+        if len(self.last_result) >= self.max_orgs:
+            return True
+
+        return False
+
+    def _get_from_cache(self) -> List[OrganizationDataVolume]:
+        """
+        Returns a batch from cache and removes the elements returned from the cache
+        """
+        if len(self.last_result) >= self.max_orgs:
+            ret_val = self.last_result[: self.max_orgs]
```

Comment:
nit: maybe `batch` is a better name than `ret_val`

Reply:
`ret_val` is a convention I've been using for many years... it represents the value that I will return from the function and I find it very useful to tell at a glance what I return.